hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/system/DabEnabledUpgradeTest.java

```diff
@@ -0,0 +1,188 @@
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sourcing;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.validateUpgradeAddressBooks;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.waitUntilStartOfNextStakingPeriod;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_BILLION_HBARS;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_MILLION_HBARS;
+import static com.hedera.services.bdd.suites.regression.system.LifecycleTest.configVersionOf;
+import static java.lang.Integer.MAX_VALUE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.HapiTestLifecycle;
+import com.hedera.services.bdd.junit.OrderedInIsolation;
+import com.hedera.services.bdd.junit.hedera.HederaNode;
+import com.hedera.services.bdd.junit.support.SpecManager;
+import com.hedera.services.bdd.spec.dsl.annotations.AccountSpec;
+import com.hedera.services.bdd.spec.dsl.entities.SpecAccount;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.SemanticVersion;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestMethodOrder;
+
+/**
+ * Asserts expected behavior of the network when upgrading with DAB enabled.
+ * <p>
+ * The test framework simulates DAB by copying the <i>config.txt</i> from the node's upgrade artifacts into their
+ * working directories, instead of regenerating a <i>config.txt</i> to match its {@link HederaNode} instances. It
+ * <p>
+ * There are three upgrades in this test. The first leaves the address book unchanged, the second removes `node1`,
+ * and the last one adds a new `node5`.
+ * <p>
+ * Halfway through the sequence, we also verify that reconnect is still possible  with only `node0` and `node2`
+ * left online while `node3` reconnects; which we accomplish by giving most of the stake to those nodes.
+ * <p>
+ * We also verify that an account staking to a deleted node cannot earn rewards.
+ * <p>
+ * See <a href="https://github.com/hashgraph/hedera-improvement-proposal/blob/main/HIP/hip-869.md#user-stories">here</a>
+ * for the associated HIP-869 user stories.
+ * <p>
+ * Since this test upgrades the software version, it must run after any other test that does a restart assuming
+ * the config version is still zero.
+ */
+@Tag(UPGRADE)
+@Order(MAX_VALUE)
+@DisplayName("Upgrading with DAB enabled")
+@HapiTestLifecycle
+@OrderedInIsolation
```

Comment:
If you could explain more of those annotations in https://github.com/hashgraph/hedera-services/blob/9c7c438fafee8eb55305f3f259acf5e97ae19a01/hedera-node/test-clients/README.md, it would be great.

Reply:
These two are mentioned [here](https://github.com/hashgraph/hedera-services/blob/14046-spec-networks-doc/hedera-node/test-clients/README.md#meta-annotations) in the README...they are basically "syntactic sugar" for standard JUnit5 annotations.